lib/ansible/modules/network/eos/eos_static_routes.py

```diff
@@ -0,0 +1,367 @@
+veos(config)#
+
+- name: Replace static route configuration
+  eos_static_routes:
+    config:
+      - vrf: testvrf
+        address_families:
+          - afi: ipv4
+            routes:
+              - dest: 165.10.1.0/24
+                next_hop:
+                  - forward_router_address: 10.1.1.2
+                    interface: "Ethernet1"
+    state: replaced
+
+After State
+-----------
+
+veos(config)#show running-config | grep "route"
+ip route 165.10.1.0/24 Ethernet1 10.1.1.2
+ip route 172.17.252.0/24 Nexthop-Group testgroup
+ip route vrf testvrf 130.1.122.0/24 Ethernet1 tag 50
+ipv6 route 2211::/64 Ethernet1 100:1::2
+ipv6 route 5001::/64 Ethernet1 50
+veos(config)#
+
+
+Before State
+-------------
+veos(config)#show running-config | grep "route"
+ip route 165.10.1.0/24 Ethernet1 10.1.1.2 100
+ipv6 route 5001::/64 Ethernet1
+veos(config)#
+
+
+- name: Gather the exisitng condiguration
+  eos_static_routes:
+    state: gathered
+
+
+returns :
+  eos_static_routes:
+    config:
+      - address_families:
+          - afi: ipv4
+            routes:
+              - dest: 165.10.1.0/24
+                next_hop:
+                  - forward_router_address: 10.1.1.2
+                    interface: "Ethernet1"
+		    admin_distance: 100
+         - afi: ipv6
+            routes:
+              - dest: 5001::/64
+                next_hop:
+                  - interface: "Ethernet1"	
+
+
+# Using rendered
+
+i  eos_static_routes:
+    config:
+      - address_families:
+          - afi: ipv4
+            routes:
+              - dest: 165.10.1.0/24
+                next_hop:
+                  - forward_router_address: 10.1.1.2
+                    interface: "Ethernet1"
+                    admin_distance: 100
+         - afi: ipv6
+            routes:
+              - dest: 5001::/64
+                next_hop:
+                  - interface: "Ethernet1"
+
+
+
+returns:
+
+ip route 165.10.1.0/24 Ethernet1 10.1.1.2 100
+ipv6 route 5001::/64 Ethernet1
+
+
+"""
+RETURN = """
+before:
+  description: The configuration prior to the model invocation.
+  returned: always
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+after:
+  description: The resulting configuration model invocation.
+  returned: when changed
+  sample: >
+    The configuration returned will always be in the same format
+     of the parameters above.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: always
+  type: list
+  sample: ['command 1', 'command 2', 'command 3']
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.eos.argspec.static_routes.static_routes import Static_routesArgs
+from ansible.module_utils.network.eos.config.static_routes.static_routes import Static_routes
+
+
+def main():
+    """
+    Main entry point for module execution
+
+    :returns: the result form module invocation
+    """
```

Comment:
you can have a check here for 'config'  when state is 'merged', 'replaced' and overridden and running config for state 'parsed'

Reply:
Done!